panoramapublic/src/org/labkey/panoramapublic/query/ExperimentAnnotationsTableInfo.java

```diff
@@ -204,11 +209,12 @@ public void renderGridCellContents(RenderContext ctx, Writer out) throws IOExcep
                 }
                 else
                 {
-                    String content = "<div><a class=\"button-small button-small-green\" style=\"margin:0px 5px 0px 2px;\""
-                                     + "href=\"\" onclick=\"showShareLink(this, '" + PageFlowUtil.filter(accessUrl) + "');return false;\""
-                               + ">Share</a>";
-                    content += "</div>";
-                    out.write(content);
+                    var link = new Link.LinkBuilder("Share")
+                            .clearClasses().addClass("button-small button-small-green")
+                            .style("margin:0px 5px 0px 2px;")
+                            .onClick("showShareLink(this, '" + PageFlowUtil.filter(accessUrl) + "');return false;");
```

Comment:
This should be OK in practice, but could break if the value being rendered includes single quotes.
```suggestion
                            .onClick("showShareLink(this, " + PageFlowUtil.jsString(accessUrl) + ");return false;");
```